Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -1455,4 +1455,10 @@ public boolean caresAboutManaColor() {
                 .filter(Objects::nonNull)
                 .anyMatch(Condition::caresAboutManaColor);
     }
+
+    public AbilityImpl copyWithZone(Zone zone) {
```

Comment:
Shouldn't this be an `Ability` rather than an `AbilityImpl`, and empty method added to the interface?

Reply:
Originally it was, but that meant I needed an implementation in `SpellAbility` too, which doesn't even have a `zone` field, so I decided the operation only makes sense on `AbilityImpl`-derived abilities